plugins/optimization-detective/detect.js

```diff
@@ -66,33 +67,53 @@ function setStorageLock( currentTime ) {
 }
 
 /**
- * Logs a message.
+ * Creates a logger object with log, warn, and error methods.
  *
- * @param {...*} message
+ * @param {boolean} [debugMode=false] - Whether to enable debug mode.
+ * @return {Logger} Logger object with log, warn, and error methods.
  */
-function log( ...message ) {
-	// eslint-disable-next-line no-console
-	console.log( consoleLogPrefix, ...message );
-}
+function createLogger( debugMode = false ) {
+	return {
+		/**
+		 * Prefix for log messages.
+		 * @type {string}
+		 */
+		prefix: '',
```

Comment:
I don't think we need to create an object here, as this will require the functions to be bound to the object to work properly. If we pass in the `prefix` then it is available here via the closure already.

```suggestion
function createLogger( debugMode = false, prefix = '' ) {
	return {
```